NEWS.md

```diff
@@ -2,9 +2,46 @@
 
 ## polars (development version)
 
+### Deprecations
+
+- The following arguments of certain LazyFrame methods, which were previously used for query optimization,
+  are deprecated in favor of the new `optimizations` argument (#1635).
+  Some arguments that were intended for internal use have been removed without deprecation.
+
+  - `type_coercion`
+  - `predicate_pushdown`
+  - `projection_pushdown`
+  - `simplify_expression`
+  - `slice_pushdown`
+  - `comm_subplan_elim`
+  - `comm_subexpr_elim`
+  - `cluster_with_columns`
+  - `no_optimization`
+  - `_type_check` (removed)
+  - `_check_order` (removed)
+  - `_eager` (removed)
+
+  Functions affected are those that gained the `optimizations` argument.
+  See the New features section below for details.
+  Also, for the experimental `<lazyframe>$sink_*` methods,
+  the above arguments are removed instead of being deprecated.
```

Comment:
I'm not a big fan of this. I know those are marked experimental, but I'd rather have a deprecation warning in those functions too. There are already so many deprecations in this PR that the marginal gain of removing those arguments entirely without a deprecation warning is very small IMO. In other words, it doesn't cost much for us to add a deprecation warning to the args in those functions, but it might prevent code from breaking.

Also, this would break the behavior in `tidypolars` and I'm not sure I could address before 1.6.0.